next.config.js

```diff
@@ -3,6 +3,14 @@ const nextConfig = {
   images: {
     domains: ["localhost:3000", "images.ctfassets.net"],
   },
+    async rewrites() {
```

Comment:
Widziałem, że w wielu miejscach polegasz na odczycie langa znajdującego się w ścieżce (lub dopisywania go do linków). Jest to ryzykowne, ale już zostańmy przy tym. Sprawę może uratować mechanizm nadpisywania ścieżek (rewrite engine). 

W momencie połączenia się z aplikacją, mechanizm ten interpretuje ścieżkę i nadpisuje ją. Aplikacja działa tak, jakby znajdowała się na tej nowej ścieżce, mimo że w przeglądarce widać nadal tą starą.

Tu dodałem najprostsze dwie linijki, które zamieniają localhost na localhost/pl. Dzięki temu poprawnie wyświetli się strona główna.

Natomiast, żeby poprawić działanie wszystkich stron, trzeba trochę skomplikować te nadpisania. Instrukcje znajdziesz tutaj:
https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites